helix-rest/src/main/java/org/apache/helix/rest/common/HelixDataAccessorWrapper.java

```diff
@@ -1,24 +1,158 @@
-import org.apache.helix.manager.zk.ZKHelixDataAccessor;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import org.apache.helix.HelixProperty;
+import org.apache.helix.PropertyKey;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZKHelixDataAccessor;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.RESTConfig;
+import org.apache.helix.rest.client.CustomRestClient;
+import org.apache.helix.rest.client.CustomRestClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * This is a wrapper for {@link ZKHelixDataAccessor} that caches the result of the batch reads it
  * performs.
  * Note that the usage of this object is valid for one REST request.
  */
 public class HelixDataAccessorWrapper extends ZKHelixDataAccessor {
+  private static final Logger LOG = LoggerFactory.getLogger(HelixDataAccessorWrapper.class);
+  private static final ExecutorService POOL = Executors.newCachedThreadPool();
+
+  private static final String PARTITION_HEALTH_KEY = "PARTITION_HEALTH";
+  private static final String IS_HEALTHY_KEY = "IS_HEALTHY";
+  private static final String EXPIRY_KEY = "EXPIRE";
+
+  private final CustomRestClient _restClient;
   private final Map<PropertyKey, HelixProperty> _propertyCache = new HashMap<>();
   private final Map<PropertyKey, List<String>> _batchNameCache = new HashMap<>();
 
   public HelixDataAccessorWrapper(ZKHelixDataAccessor dataAccessor) {
     super(dataAccessor);
+    _restClient = CustomRestClientFactory.get();
+  }
+
+  HelixDataAccessorWrapper(ZKHelixDataAccessor dataAccessor, CustomRestClient client) {
+    super(dataAccessor);
+    _restClient = client;
+  }
+
+  public List<ExternalView> getExternalViews() {
+    return getChildNames(keyBuilder().externalViews()).stream()
```

Comment:
You dont have to read it in that way. This causes more read request to ZK. It should be getChildValues(keyBuikder().externalViews()). If you would like to memorize it memorize it in a different way.